nncf/tensor/functions/tf_numeric.py

```diff
@@ -0,0 +1,534 @@
+
+
+@numeric.mean.register(tf.Tensor)
+def _(
+    a: tf.Tensor,
+    axis: Union[int, Tuple[int, ...]] = None,
+    keepdims: bool = False,
+    dtype: Optional[TensorDataType] = None,
+) -> tf.Tensor:
+    with tf.device(a.device):
+        return tf.reduce_mean(a, axis=axis, keepdims=keepdims)
+
+
+@numeric.median.register(tf.Tensor)
+def _(
+    a: tf.Tensor,
+    axis: Union[int, Tuple[int, ...]] = None,
+    keepdims: bool = False,
+) -> tf.Tensor:
+    numpy_a = np.array(a)
+    numpy_median = np.median(numpy_a, axis=axis, keepdims=keepdims)
+
+    with tf.device(a.device):
+        tf_median = tf.constant(numpy_median)
+
+    return tf_median
+
+
+@numeric.round.register(tf.Tensor)
+def _(a: tf.Tensor, decimals=0) -> tf.Tensor:
+    scale_factor = 10**decimals
+    scaled_tensor = a * scale_factor
+    with tf.device(a.device):
+        rounded_tensor = tf.round(scaled_tensor)
+        return rounded_tensor / scale_factor
+
+
+@numeric.power.register(tf.Tensor)
+def _(a: tf.Tensor, exponent: Union[tf.Tensor, float]) -> tf.Tensor:
+    with tf.device(a.device):
+        return tf.pow(a, exponent)
+
+
+@numeric.quantile.register(tf.Tensor)
+def quantile(
+    a: tf.Tensor,
+    q: Union[float, List[float]],
+    axis: Optional[Union[int, Tuple[int]]] = None,
+    keepdims: bool = False,
+) -> tf.Tensor:
+    a_np = a.numpy()
+    quantile_np = np.quantile(a_np, q=q, axis=axis, keepdims=keepdims)
+    with tf.device(a.device):
+        return tf.constant(quantile_np)
+
+
+@numeric.percentile.register(tf.Tensor)
+def _(
+    a: tf.Tensor,
+    q: Union[float, List[float]],
+    axis: Union[int, Tuple[int, ...], List[int]],
+    keepdims: bool = False,
+) -> List[Union[tf.Tensor, np.generic]]:
+    with tf.device(a.device):
+        q = [x / 100 for x in q] if isinstance(q, (list, tuple)) else q / 100
+        return numeric.quantile(a, q=q, axis=axis, keepdims=keepdims)
+
+
+@numeric._binary_op_nowarn.register(tf.Tensor)
+def _(a: tf.Tensor, b: Union[tf.Tensor, float], operator_fn: Callable) -> tf.Tensor:
+    with tf.device(a.device):
+        return operator_fn(a, b)
+
+
+@numeric._binary_reverse_op_nowarn.register(tf.Tensor)
+def _(a: tf.Tensor, b: Union[tf.Tensor, float], operator_fn: Callable) -> tf.Tensor:
+    with tf.device(a.device):
+        return operator_fn(b, a)
+
+
+@numeric.clip.register(tf.Tensor)
+def _(a: tf.Tensor, a_min: Union[tf.Tensor, float], a_max: Union[tf.Tensor, float]) -> tf.Tensor:
+    with tf.device(a.device):
+        return tf.clip_by_value(a, a_min, a_max)
+
+
+@numeric.finfo.register(tf.Tensor)
+def _(a: tf.Tensor) -> TypeInfo:
+    ti = tf.experimental.numpy.finfo(a.dtype)
+    return TypeInfo(ti.eps, ti.max, ti.min)
+
+
+@numeric.as_tensor_like.register(tf.Tensor)
+def _(a: tf.Tensor, data: Any) -> tf.Tensor:
+    with tf.device(a.device):
+        return tf.convert_to_tensor(data)
+
+
+@numeric.item.register(tf.Tensor)
+def _(a: tf.Tensor) -> Union[int, float, bool]:
```

Comment:
it still returns a tensor (with empty shape) and the following logic is still necessary. but the addition of reshape here allows to avoid deprecation warning, so, I'll add it. thanks for pointing out.